backend/dataall/api/Objects/DatasetProfiling/resolvers.py

```diff
@@ -89,6 +90,21 @@ def get_profiling_run(context: Context, source, profilingRunUri=None):
 
 def get_last_table_profiling_run(context: Context, source, tableUri=None):
     with context.engine.scoped_session() as session:
+        table: models.DatasetTable = db.api.DatasetTable.get_dataset_table_by_uri(
```

Comment:
I wonder if we should start profiling for secret datasets at all?

If so, we should add more checks in `list_table_profiling_runs` , `get_profiling_run`, `list_profiling_runs`, ( + `queue_profiling_run`, `update_profiling_run_results` also don't have any permission check) 😃

Reply:
@nikpodsh thanks for reviewing! You are right on reviewing this. I have taken a closer look at the resolvers of DataserProfiling. 
- I deleted those that were not used in the frontend: `update_profiling_run_results`, `get_profiling_run`, `list_profiling_runs`
- Added check permissions in `list_table_profiling_runs` I think you are right, enforcing permission checks is always a good enhacement.
- Renamed `get_profiling_results` --> `resolve_profiling_results` and `resolve_profiling_run_status` to remark that they are part of the Profiling schema and not APIs
- Renamed `get_profiling_results_from_s3` to `_get_profiling_results_from_s3` to indicate that it is an internally used function.
- Some that you mention I was not able to find: `queue_profiling_run`, `update_profiling_run_results`